src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxButton.java

```diff
@@ -233,4 +255,26 @@ protected void doRendererPaint(final Graphics g, final ButtonModel buttonModel,
         // Remove component from renderer pane, allowing it to be gc'ed.
         rendererPane.remove(c);
     }
+
+    @Override
+    public AccessibleContext getAccessibleContext() {
+        if (accessibleContext == null) {
+            accessibleContext = new AccessibleAquaComboBoxButton();
+        }
+        return accessibleContext;
+    }
+
+    private final class AccessibleAquaComboBoxButton extends AccessibleJButton {
+        @Override
+        public String getAccessibleName() {
+            String name = super.getAccessibleName();
+            if (name == null || name.isEmpty()) {
+                if (!comboBox.isEditable() && comboBox.getSelectedItem() != null) {
+                    Component c = getRendererComponent();
+                    name = c.getAccessibleContext().getAccessibleName();
+                }
+            }
```

Comment:
Conditions combined.

> Since the arrow button is an internal component, we may drop calling super. However, I'm unsure about it.

I think we can keep this as it is.